src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -3083,6 +3083,7 @@ private void setRelays(List<String> allRelays)
 
             OctoParticipant octoParticipant = getOctoParticipant(remoteRelays);
             octoParticipant.setRelays(remoteRelays);
+            updateColibriOctoChannels(octoParticipant);
```

Comment:
I am not sure that it is safe to call `updateColibriOctoChannels` without checking whether the "session" has been established (i.e. jicofo has learnt the IDs of the colibri channels). The existing code always does that check.

We already have code which will [update the relay IDs if they changed before the session was established](https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/OctoChannelAllocator.java#L165), so I think you can just wrap the code in `if (octoParticipant.isSessionEstablished())` and do nothing if the session is not established.

Reply:
Cool, I can add that check